resource/config.go

```diff
@@ -28,10 +28,12 @@ type Config struct {
 	AssociatedAttributes      map[Name]AssociatedConfig
 	ConvertedAttributes       ConfigValidator
 	ImplicitDependsOn         []string
+	ImplicitOptionalDependsOn []string
```

Comment:
Implicit _optional_ dependencies (`ImplicitOptionalDependsOn`) are stored on resource configs in an identical way to implicit _required_ dependencies (`ImplicitDependsOn`) (including "caching" upon validation).